Dockerfile

```diff
@@ -1,28 +1,43 @@
-FROM gcr.io/world-fishing-827/github.com/globalfishingwatch/gfw-bash-pipeline:latest-python3.8
+# ---------------------------------------------------------------------------------------
+# BASE
+# ---------------------------------------------------------------------------------------
+FROM gcr.io/world-fishing-827/github.com/globalfishingwatch/gfw-bash-pipeline:latest-python3.8 as base
+
+# Copy files from official SDK image, including script/dependencies.
+COPY --from=apache/beam_python3.8_sdk:2.56.0 /opt/apache/beam /opt/apache/beam
 
 # Install SDK. (needed for Python SDK)
 RUN pip install --no-cache-dir apache-beam[gcp]==2.56.0
 
-# Copy files from official SDK image, including script/dependencies.
-COPY --from=apache/beam_python3.8_sdk:2.56.0 /opt/apache/beam /opt/apache/beam
+# Install application dependencies
+COPY requirements.txt /opt/requirements.txt
+RUN pip install --no-cache-dir -r /opt/requirements.txt
 
-# Perform any additional customizations if desired
-COPY ./requirements.txt ./
-RUN pip install --no-cache-dir -r requirements.txt
+# Set the entrypoint to Apache Beam SDK launcher.
+ENTRYPOINT ["/opt/apache/beam/boot"]
 
-# Temporary. TODO: Use a local test docker image with extra dependencies.
-COPY ./requirements/test.txt ./
-RUN pip install --no-cache-dir -r test.txt
+# ---------------------------------------------------------------------------------------
+# PROD
+# ---------------------------------------------------------------------------------------
+FROM base as prod
+
+# Install app package
+COPY . /opt/project
+RUN pip install .
+
+# ---------------------------------------------------------------------------------------
+# DEV
+# ---------------------------------------------------------------------------------------
+FROM base as dev
 
-# Temporary. TODO: Use a local dev docker image with extra dependencies.
 COPY ./requirements/dev.txt ./
+COPY ./requirements/test.txt ./
+
 RUN pip install --no-cache-dir -r dev.txt
+RUN pip install --no-cache-dir -r test.txt
 
-# Setup local packages
+# Install app package
 COPY . /opt/project
 ENV PYTHONPATH /opt/project
 RUN cd /usr/local/lib/python3.8/site-packages && \
     python /opt/project/setup.py develop
-
-# Set the entrypoint to Apache Beam SDK launcher.
-ENTRYPOINT ["/opt/apache/beam/boot"]
```

Comment:
Beam boot should be set as entrypoint only for prod

Reply:
Now that I think about it, I think it makes sense to have this entrypoint in the section in which we are installing apache beam system dependencies...But what we can do is to modify the entry point in the dev target. What do you think?